zhaquirks/tuya/ts601_door.py

```diff
@@ -0,0 +1,230 @@
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+    ZONE_STATE,
+)
+from zhaquirks.tuya import (
+    TUYA_CLUSTER_ID,
+    DPToAttributeMapping,
+    TuyaLocalCluster,
+    TuyaNewManufCluster,
+)
+
+DOOR_HANDLE_DP_ID = 1
+BATTERY_STATE_DP_ID = 3
+VIBRATION_DP_ID = 10
+
+DP_HANDLER_EP_ID = 1
+DOOR_HANDLE_EP_ID = 2
+VIBRATION_EP_ID = 3
+
+
+class CustomBasicCluster(CustomCluster, Basic):
+    """Custom Basic cluster to report power source."""
+
+    _CONSTANT_ATTRIBUTES = {
+        Basic.AttributeDefs.power_source.id: Basic.PowerSource.Battery
+    }
+
+
+class CustomTuyaPowerConfigurationCluster(TuyaLocalCluster, PowerConfiguration):
+    """Custom Power Configuration cluster that represents battery reports."""
+
+    _CONSTANT_ATTRIBUTES = {
+        PowerConfiguration.AttributeDefs.battery_size.id: PowerConfiguration.BatterySize.AAA,
+        PowerConfiguration.AttributeDefs.battery_quantity.id: 2,
+    }
+
+    def __init__(self, endpoint: Endpoint, is_server: bool = True):
+        """Init constructor that initializes battery attributes."""
+        super().__init__(endpoint=endpoint, is_server=is_server)
+
+        # Initialize remaining battery to 100% until a proper measurement arrives.
+        # Device measures battery in 1% steps, while the ZCL specifies it in 0.5% steps.
+        self.update_attribute(
+            attr_name=PowerConfiguration.AttributeDefs.battery_percentage_remaining.name,
+            value=100 * 2,
+        )
+
+
+class CustomTuyaContactSwitchCluster(TuyaLocalCluster, IasZone):
+    """Custom IasZone cluster that represents the Open/Closed sensor."""
+
+    _CONSTANT_ATTRIBUTES = {
+        IasZone.AttributeDefs.zone_state.id: ZONE_STATE,
+        IasZone.AttributeDefs.zone_type.id: ZoneType.Contact_Switch,
+    }
+
+    def __init__(self, endpoint: Endpoint, is_server: bool = True):
+        """Init constructor that initializes the major IasZone attributes."""
+        super().__init__(endpoint=endpoint, is_server=is_server)
+
+        # Initialize zone status until a proper update comes.
+        self.update_attribute(
+            attr_name=IasZone.AttributeDefs.zone_status.name,
+            value=0x0000,
+        )
+
+
+class CustomTuyaVibrationCluster(TuyaLocalCluster, IasZone):
+    """Custom IasZone cluster that represents the vibration sensor."""
+
+    _CONSTANT_ATTRIBUTES = {
+        IasZone.AttributeDefs.zone_state.id: ZONE_STATE,
+        IasZone.AttributeDefs.zone_type.id: ZoneType.Vibration_Movement_Sensor,
+    }
+
+    def __init__(self, endpoint: Endpoint, is_server: bool = True):
+        """Init constructor that initializes the major IasZone attributes."""
+        super().__init__(endpoint=endpoint, is_server=is_server)
+
+        # Initialize zone status until a proper update comes.
+        self.update_attribute(
+            attr_name=IasZone.AttributeDefs.zone_status.name,
+            value=0x0000,
+        )
```

Comment:
Same thing here.

Reply:
Removed for now.

Reply:
This is where I disagree. Significant force is very rarely applied to most doors (in fact, my sensor only detected such change when I was manually testing the quirk, but never since then). This leads to an unitialized entity for quite a long time, which I personally don't really like to see on a UI. Initializing the vibration state to "no vibration" seems to me the lesser evil :)